### PR TITLE
add option to use losses either from torch.nn or from torch.nn.functional

### PR DESCRIPTION
**Pull request by bruce-willis**
Now it should be easier to play around with different losses available in the `torch` library.  Instead of writing wrappers, it is possible just to provide the name of a loss (e.g., `MSELoss` instead of `fsaa/losses/mse_loss.py` file) 

For the example in the `readme` file, mse in the feature domain is increased by 0.4 and cosine similarity is decreased by 0.007 by utilizing `SmoothL1Loss` (which is not a significant improvement, but more comprehensive testing is needed).

P.S. I couldn't make `pre-commit` hooks work :(

### CODE REVIEW

**Comment by BrianPulfer** [comment]
Hi @bruce-willis! Thanks for the PR, will look into it ASAP.

For the pre-commit hooks, `autopep8` has to be installed locally in your environment with `pip install autopep8`. This probably needs to be improved to just use the remote repo.

File: tests/test_losses.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+def data():
+    shape = (1, 3, 224, 224)
+    return torch.randn(shape).clamp(0, 1), torch.randn(shape).clamp(0, 1)
+
+
+def test_custom_losses():
+    """Tests that the all custom losses return a loss"""
+    for loss_name in LOSSES:
+        loss_fn = get_loss(loss_name)
+        assert callable(loss_fn)
```
Comment (BrianPulfer) [inline tests/test_losses.py:19]: All `nn.Module` objects are callable, so I would just check assert `isinstance(loss_fn, nn.Module)`.

**Review by BrianPulfer — COMMENTED** [review]
The `get_loss` method was initially put as a utility to easily get loss functions that are useful in the context of adversarial attacks but that are not already implemented in `nn` and `nn.functional` (like LPIPS).

`nn` modules can be directly passed to `fsaa.attack.attack` method.

e.g.

```python
import torch.nn as nn
from fsaa.attack import attack

batch = ...
model = ...
adv_batch = attack(
    model,
    batch,
    image_loss=nn.MSELoss()
)
```
Currently, MSE and Cosine Similarity losses are supported in `get_loss` just for completeness, but it might be better to actually remove them in future.

**Review by BrianPulfer — APPROVED** [review]
Great! Thank you for your contribution. This solves #2 and #4!
